iidm/iidm-converter-api/src/main/java/eu/itesla_project/iidm/datasource/ZipMemDataSource.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2016, All partners of the iTesla project (http://www.itesla-project.eu/consortium)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package eu.itesla_project.iidm.datasource;
+
+import eu.itesla_project.commons.io.ForwardingInputStream;
+import eu.itesla_project.commons.io.ForwardingOutputStream;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * @author Giovanni Ferrari <giovanni.ferrari@techrain.it>
+ */
+public class ZipMemDataSource extends ReadOnlyMemDataSource {
+
+    public ZipMemDataSource() {
+        super();
+    }
+    
+    public ZipMemDataSource(byte[] content) {
+        super();
+        Objects.requireNonNull(content);
+        InputStream is = new ByteArrayInputStream(content);
+       
+        try( ZipInputStream zipStream = new ZipInputStream(is)) {
+            ZipEntry entry = zipStream.getNextEntry();
+            if(entry != null)
+            {
+                String entryname = entry.getName();
+                String format = entryname.substring(entryname.lastIndexOf(".")+1);
+                putData(null, format, content);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
It seems this code is not working very well: for each entry, the full content of the zip file is stored in the data map. Maybe you have to unzip the bytes for each file.